monai/transforms/utils.py

```diff
@@ -587,6 +587,7 @@ def generate_label_classes_crop_centers(
     num_samples: int,
     label_spatial_shape: Sequence[int],
     indices: Sequence[NdarrayOrTensor],
+    img_spatial_shape: Sequence[int] | None = None,
```

Comment:
I still don't think we should add `img_spatial_shape` here. We should assume image and label have the same shape, so a warning should be enough.